From 802ed5ce00562d765a1d799bdbadba4664bee128 Mon Sep 17 00:00:00 2001
From: Nathan Pratta Teodosio <nathan.teodosio@canonical.com>
Date: Mon, 26 Sep 2022 14:41:21 -0300
Subject: [PATCH] Allow {get,set}_mempolicy syscalls

This fixes builds that fail with 
"Disallowed syscall "get_mempolicy" caught in sandbox"

Bug-Upstream: https://gitlab.gnome.org/GNOME/tracker-miners/-/issues/234

---
 src/libtracker-miners-common/tracker-seccomp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libtracker-miners-common/tracker-seccomp.c b/src/libtracker-miners-common/tracker-seccomp.c
index 88fbbe197..cac712f37 100644
--- a/src/libtracker-miners-common/tracker-seccomp.c
+++ b/src/libtracker-miners-common/tracker-seccomp.c
@@ -102,6 +102,8 @@ tracker_seccomp_init (void)
 
 	/* Memory management */
 	ALLOW_RULE (brk);
+	ALLOW_RULE (get_mem_policy);
+	ALLOW_RULE (set_mem_policy);
 	ALLOW_RULE (mmap);
 	ALLOW_RULE (mmap2);
 	ALLOW_RULE (munmap);
-- 
2.37.2

